### PR TITLE
fix(a380x): FMS+ROP/ROW fixes

### PR DESCRIPTION
**Pull request by flogross89**
## Summary of Changes

- Performance optimization: Less allocations for TypeScript BTV code, replace Arinc429Word with Arinc429Register
- Even if Navigraph sub available, set landing runway from FMS flight plan at 300ft RA (if not set via OANS before)
- Fix display of fuel predictions, overall fixes for FUEL&LOAD page

## Additional context

Discord username (if different from GitHub):

## Testing instructions

1. Load FPLN, verify FUEL&LOAD page somewhat makes sense and EFOB predictions are showing on the FPLN page. Furthermore, DEST ETA should now also be filled correctly
2. Perform landing at way too short runway with Navigraph connected and OANS working, but don't select any landing runway via the OANS interface, just via the FMS. At 300ft RA, "RUNWAY TOO SHORT" should be audible. Land anyway with ABRK selected to LO, or delay braking until the end of the runway. ROP should now activate, with "MAX BRAKING" and/or "MAX REVERSE". 
3. Perform landing with BTV set via OANS (with Navigraph connected). Everything should happen as usual, BTV should NOT disarm when passing 300ft RA

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

**Comment by flogross89** [comment]
only merge after https://github.com/flybywiresim/aircraft/pull/9063 to avoid conflicts

**Review by 2hwk — APPROVED** [review]

File: fbw-common/src/systems/instruments/src/OANC/BtvPublisher.ts [hunk 1]
```diff
@@ -74,21 +83,24 @@ export class BtvArincProvider implements Instrument {
...
+        publisher.pub('btvRot', this.btvRot);
       });
 
     subscriber
       .on('btvTurnAroundIdleReverseRaw')
       .whenChanged()
       .handle((w) => {
-        publisher.pub('btvTurnAroundIdleReverse', new Arinc429Word(w));
+        this.btvTurnAroundIdleReverse.set(w);
+        publisher.pub('btvTurnAroundIdleReverse', this.btvTurnAroundIdleReverse);
```
[the remark this answers is not in the document]
Comment (flogross89) [reply fbw-common/src/systems/instruments/src/OANC/BtvPublisher.ts:95]: Do you mean Arinc429LocalVarConsumerSubject? I tried overhauling it in my latest commit

File: fbw-common/src/systems/instruments/src/OANC/Oanc.tsx [hunk 2]
```diff
@@ -911,8 +911,13 @@ export class Oanc<T extends number> extends DisplayComponent<OancProps<T>> {
...
   private lastTime = 0;
 
-  private projectCoordinates(coordinates: Coordinates): [number, number] {
-    return globalToAirportCoordinates(this.arpCoordinates, coordinates);
+  /**
+   *
+   * @param projectedCoordinates Output argument: Write projected coordinates here
+   * @param coordinates coordinates to be transformed
+   */
+  private projectCoordinates(projectedCoordinates: Position, coordinates: Coordinates) {
```
Comment (tracernz) [inline fbw-common/src/systems/instruments/src/OANC/Oanc.tsx:962]: Better to name it out to make it clear that it's the out param, and it's also nice to return it so you can chain calls, or put it in a handy local scoped name if you're using a cache object. Also nice to put the out param at the end by convention.

**Review by tracernz — APPROVED** [review]
Would be good to fix per comment, but otherwise okay to go.

File: fbw-common/src/systems/instruments/src/OANC/Oanc.tsx (hunk 2 above)
Comment (flogross89) [reply fbw-common/src/systems/instruments/src/OANC/Oanc.tsx:962]: done

**Review by tracernz — APPROVED** [review]
💯

**Comment by BlueberryKing** [comment]
Dev Team Report

Discord Username : BlueberryKing
Object of testing : https://github.com/flybywiresim/aircraft/pull/9069
Aircraft : A380X
Tier of Testing : 1
Date : 19/10/2024

Testing Process:

- Load up at NZWN on the runway, set up FMS
- Fly NZNS RNP 02 approach
- Observe RUNWAY TOO SHORT below 300 ft
- Land without braking
- Observe SET MAX REVERSE, MAX BRAKING callouts
- Set up on approach to NZAA 05R, set up BTV
- Observe BTV disarmed at around 500 ft, go around, try again
- BTV works

Testing Results:
Passed

Negatives:
- SET MAX REVERSE, MAX BRAKING callouts overlap each other
- BTV disamed itself on the first attempt and I don't know why, but it worked well the second time.

Conclusions:
Passed

Media:
N/A

**Comment by flogross89** [comment]
> Dev Team Report
> 
> Discord Username : BlueberryKing Object of testing : #9069 Aircraft : A380X Tier of Testing : 1 Date : 19/10/2024
> 
> Testing Process:
> 
> * Load up at NZWN on the runway, set up FMS
> * Fly NZNS RNP 02 approach
> * Observe RUNWAY TOO SHORT below 300 ft
> * Land without braking
> * Observe SET MAX REVERSE, MAX BRAKING callouts
> * Set up on approach to NZAA 05R, set up BTV
> * Observe BTV disarmed at around 500 ft, go around, try again
> * BTV works
> 
> Testing Results: Passed
> 
> Negatives:
> 
> * SET MAX REVERSE, MAX BRAKING callouts overlap each other
> * BTV disamed itself on the first attempt and I don't know why, but it worked well the second time.
> 
> Conclusions: Passed
> 
> Media: N/A

Thanks! We don't have proper sound scheduling yet, so the overlapping can't be fixed without a callout rework :/
And for the disarming, we have to monitor that, there are quite a few reasons why it disarms on approach. But let's hope it doesn't systematically disarm on approach.

**Pull request merged**